runconfig/hostconfig_windows.go

```diff
@@ -26,9 +26,41 @@ func ValidateNetMode(c *container.Config, hc *container.HostConfig) error {
 		return nil
 	}
 	parts := strings.Split(string(hc.NetworkMode), ":")
-	if len(parts) > 1 {
+	if len(parts) > 2 {
 		return fmt.Errorf("invalid --net: %s", hc.NetworkMode)
 	}
+	if hc.NetworkMode.IsContainer() {
```

Comment:
All those checks are common with the `unix` side. We should re-use them somehow instead of duplicating (it makes it harder to maintain if we duplicate).